src/Stratis.Features.Diagnostic/Controllers/DiagnosticController.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Controllers;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Stratis.Features.Diagnostic.Controllers.Models;
+using Stratis.Features.Diagnostic.PeerDiagnostic;
+using Stratis.Features.Diagnostic.Utils;
+
+namespace Stratis.Features.Diagnostic.Controllers
+{
+    /// <summary>
+    /// Controller providing operations on a wallet.
+    /// </summary>
+    [Route("api/[controller]/[action]")]
+    public class DiagnosticController : FeatureController
+    {
+        private readonly PeerStatisticsCollector peerStatisticsCollector;
+
+        public DiagnosticController(PeerStatisticsCollector peerDiagnosticCollector, IConnectionManager connectionManager, IConsensusManager consensusManager)
+            : base(connectionManager: connectionManager, consensusManager: consensusManager)
+        {
+            this.peerStatisticsCollector = peerDiagnosticCollector;
+        }
+
+        /// <summary>
+        /// Returns the connected peers with some information
+        /// </summary>
+        [HttpGet]
+        public IActionResult GetConnectedPeersInfo()
+        {
+            try
+            {
+                var peersByPeerId = this.ConsensusManager.GetPrivateFieldValue<Dictionary<int, INetworkPeer>>("peersByPeerId").Values.ToList();
+                var connectedPeers = this.ConnectionManager.ConnectedPeers.ToList();
+
+                object DumpPeer(INetworkPeer peer)
+                {
+                    return new { peer.IsConnected, peer.DisconnectReason, peer.State, EndPoint = peer.PeerEndPoint.ToString() };
+                }
+
+                return this.Json(new
+                {
+                    peersByPeerId = peersByPeerId.Select(DumpPeer),
+                    connectedPeers = connectedPeers.Select(DumpPeer),
+                    connectedPeersNotInPeersByPeerId = connectedPeers.Except(peersByPeerId).Select(DumpPeer)
```

Comment:
Is it possible to have connected peers in the connection manager that is not in ConsensusManager's peer collection?

Reply:
I remember I created this GetConnectedPeersInfo a couple of month ago because this was a condition that was happening sometimes, I don't know if actually it can still happen. Basically I had inconsistent console output where I was reporting to be collected multiple times with the same peer, or not connected to a peer in console but actually I was sending data to him anyway.
Being this a diagnostic feature this method may be useful when we have awkward thing happening, but as I said I'm not sure if this may still happen